### DISCUSSION TITLE
How to specify the path of the firejail.config file？

### DISCUSSION THREAD

**Discussion by smartfind** [netblue30/firejail#6554] [Q&A] [answered]
I have a need for migration in my business, but I have found that once Firejail is compiled, I cannot re specify the Firejail.config path, resulting in the Firejail.config file not being found. Is there any good solution?

**Comment by rusty-snake** [comment]
- re-compile
- symlink
- submit patch and wait for release

**Reply by smartfind** [reply]
Hello, is there currently a direct way to specify the firejail.config path?

**Reply by rusty-snake** [reply]
No:

https://github.com/netblue30/firejail/blob/66a281875ac81691f70b2fcd6d7582dc319ede36/src/firejail/checkcfg.c#L68

Also what do you mean with "to specify the firejail.config path" more precisely? i.e. where/how to specify.

**Reply by smartfind** [reply]
After compiling in Firejail, I need to migrate to another environment for use. If the path stored in the migration environment is different from the path output by the compilation, there will be an error message saying that Firejail.config cannot be found. At this time, I hope to specify the Firejail.config path through a method similar to -- config.

**Reply by rusty-snake** [reply, marked as answer]
Implementing a command-line switch for the path to firejail.config sound nearly impossible to implement safely.

**Reply by smartfind** [reply]
Well, thank you very much. I do have a need to migrate to any target environment and call any directory here. I will think of other solutions.

**Reply by kmk3** [reply]
> Well, thank you very much. I do have a need to migrate to any target
> environment and call any directory here. I will think of other solutions.

While this specific use case may be difficult to support safely, it would still
be helpful to know what kind of environments firejail is being used in, to keep
them in mind when considering the tradeoffs of bumping the minimum supported
version of the kernel/distro/tools/etc.

Would you mind specifying the following for the host and target (taken from
[`build_issue.md`][1])?

- Name/version/arch of the Linux kernel (`uname -srm`):
- Name/version of the Linux distribution (e.g. "Ubuntu 20.04" or "Arch Linux"):
- Name/version of the C compiler (e.g. "gcc 14.1.1-1"):
- Name/version of the libc (e.g. "glibc 2.40-1"):
- Name/version of the Linux API headers (e.g. "linux-api-headers 6.10-1" on
  Arch Linux):

See also:

* https://github.com/netblue30/firejail/wiki/Firejail-Universe

[1]: https://github.com/netblue30/firejail/blob/508394fbd89320a36af23c982c9676884cfa5fa1/.github/ISSUE_TEMPLATE/build_issue.md#environment
